### PR TITLE
Fix SQL injection in List filters

### PR DESCRIPTION
**Pull request by iaroslav-ciupin**
# TL;DR
Validate filters in List API endpoints so that they only contain supported columns and columns of joined tables. 
SQL injection of sort keys is done in separate [PR](https://github.com/flyteorg/flyteadmin/pull/606) 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: pkg/manager/impl/util/filters.go [hunk 1]
```diff
@@ -120,6 +120,54 @@ func prepareValues(field string, values []string) (interface{}, error) {
 	return preparedValues, nil
 }
 
+var allowedJoinTableColumns = map[common.Entity]map[common.Entity]sets.String{
+	common.Execution: {
+		common.Execution:  gormimpl.ExecutionColumns,
```
Comment (katrogan) [inline pkg/manager/impl/util/filters.go:125]: instead of duplicating the mapping between common.Entity -> allowedcolumns, can we create two sets here?
1) for entity -> allowed join entities
2) for entity -> allowed columns
